homeassistant/helpers/entity_values.py

```diff
@@ -0,0 +1,42 @@
+"""A class to hold entity values."""
+import fnmatch
+import re
+
+from homeassistant.core import split_entity_id
+
+
+class EntityValues(object):
+    """Class to store entity id based values."""
+
+    def __init__(self, exact=None, domain=None, glob=None):
+        """Initialize an EntityConfigDict."""
+        self._cache = {}
+        self._exact = exact
+        self._domain = domain
+
+        if glob is not None:
+            glob = {re.compile(fnmatch.translate(key)): value
+                    for key, value in glob.items()}
+
+        self._glob = glob
+
+    def get(self, entity_id):
```

Comment:
1) You dropped deep update - please bring it back.
The usecase is setting a config dict for domain and overriding single entities for specific deives

2) You switched order to make domain stronger than glob. I think a more common usecase is the opposite. (Except maybe `*` glob to set some global default, which I'm not sure is a real usecase.